e2e-tests/spring-rest-mongo/src/main/java/com/mongo/reservations/ReservationsController.java

```diff
@@ -0,0 +1,21 @@
+package com.mongo.reservations;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping(path = "/reservations")
+public class ReservationsController {
+
+    @Autowired
+    private ReservationRepository reservations;
+
+    @GetMapping("findAll")
+    public ResponseEntity<Void> findAll() {
+        int status = (reservations.findAll().isEmpty()) ? 400 : 200 ;
```

Comment:
shouldn't we explicitely check that at least one `checkInDateTime` is not null?

Reply:
ok. I will add a @NotNull annotation plus some logic in the controller